script.js

```diff
@@ -1,4 +1,124 @@
+  return `Please input '1' or '2' to choose the dice's order. <br><br> Your dice rolls are: <br>Dice1: ${currentPlayerRolls[0]} & Dice2: ${currentPlayerRolls[1]}. `
+  } 
+  if (playerInput == 1) {
+    console.log('Control flow: playerInput == 1');
+
+    playerScore = Number(String(currentPlayerRolls[0])+ String(currentPlayerRolls[1]));
+    allPlayersScore.push(playerScore);
+   currentPlayerRolls = [];
+
+  return `PLAYER ${currentPlayer}<br> Your chosen value is: ${playerScore}.`
+  }
+  if (playerInput == 2){
+   console.log('Control flow: playerInput == 2');
+
+    playerScore = Number(String(currentPlayerRolls[1])+ String(currentPlayerRolls[0]));
+  allPlayersScore.push(playerScore);
+  currentPlayerRolls = [];
+
+  return `PLAYER ${currentPlayer} <br> Your chosen value is: ${playerScore}.`
+  }
+}
+
+var comparePlayerScores = function(){
+  var compareMessage = `Player 1 score: ${allPlayersScore[0]}<br> Player 2 score: ${allPlayersScore[1]}`
+ if (allPlayersScore[0] > allPlayersScore[1]){
+    compareMessage += `Player 1 wins.👩`
+  };
+  if (allPlayersScore[0] < allPlayersScore[1]){
+    compareMessage +=`Player 2 wins.🧑`
+  };
+    if (allPlayersScore[0] == allPlayersScore[1]){
+    compareMessage +=`This is a tie.🤷‍♀️`
+  };
+  return compareMessage;
+}
+
+var restGame = function(){
+  currentPlayer = 1;
+  gameState = GAME_STATE_DICE_ROLL;
+  allPlayersScore = [];
+}
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log('Checking game state on submit click: ', gameState)
+  console.log('Checking currentPlayer on submit click: ', currentPlayer);
+
+  var outputMessage = '';
+ if(gameState == GAME_STATE_DICE_ROLL){
+  console.log('Control flow: gameState == GAME_STATE_DICE_ROLL');
+
+gameState = GAME_STATE_CHOOSE_DICE_ORDER;
+outputMessage = rollDiceForPlayer();
+return outputMessage;
+ }
+ if(gameState == GAME_STATE_CHOOSE_DICE_ORDER){
+  console.log("Control flow: gameState == GAME_STATE_CHOOSE_DICE_ORDER");
+
+  outputMessage = getPlayerScore(input);
+ 
+  
+  if(currentPlayer == 1){
+    console.log("control flow: end of player1 turns,now player2 turns");
+
+    currentPlayer = 2;
+    gameState = GAME_STATE_DICE_ROLL;
+    return `${outputMessage} <br><br> Now is the Player2 turn & Click the submit button.`; 
+  }
+  if(currentPlayer == 2){
+    console.log("control flow: end of player2 turn and submit click will calculate score");
+
+    gameState = GAME_STATE_COMPARE_SCORE; 
+  return `${outputMessage} <br><br> Press submit to calculate scores.`;
 };
+};
+if (gameState == GAME_STATE_COMPARE_SCORE){
+  console.log('control flow:game state== GAME_STATE_COMPARE_SCORES');
+
+  outputMessage = comparePlayerScores();
+  
+  restGame();
+  console.log('current player after rest : ', currentPlayer);
+  console.log('game state reset :',gameState);
+  console.log('allPlayersScoreg array:', allPlayersScore);
+  
+  return outputMessage;
+};
+};
```

Comment:
Main function is relatively concise and readable, with most of the logic abstracted away to helper functions! There are some issues with indentation, you might want to look into indenting your code to format it nicely :)

Remember to use ELSE IF as well, to imply that only 1 code block can run at any one time.